core/src/test/scala/zio/config/SetTest.scala

```diff
@@ -271,6 +271,36 @@ object SetTest
             )
           )
           assert(res)(isLeft(equalTo(expected)))
+        },
+        test("accumulates all errors") {
+          case class Cfg(a: Set[Boolean], b: Set[Int])
+
+          val cCfg = (nested("a")(setStrict(boolean)) |@| nested("b")(setStrict(int)))(Cfg, Cfg.unapply)
+
+          val res = read(
+            cCfg from ConfigSource.fromPropertyTree(
+              Record(
+                Map(
+                  "a" -> Sequence(Leaf("true") :: Leaf("lorem ipsum") :: Nil),
+                  "b" -> Sequence(Leaf("one") :: Leaf("2") :: Nil)
+                )
+              ),
+              "tree"
+            )
+          )
+          val expected: ReadError[String] =
+            AndErrors(
+              List(
+                AndErrors(
+                  List(
+                    FormatError(List(Key("a"), Index(1)), "Provided value is lorem ipsum, expecting the type boolean")
+                  )
+                ),
+                AndErrors(List(FormatError(List(Key("b"), Index(0)), "Provided value is one, expecting the type int")))
+              )
+            )
+
+          assert(res)(isLeft(equalTo(expected)))
```

Comment:
Let's test whether index associated with errors test separately.
For list and map

Also, great to have another complicated case class test with all of them failed.

Reply:
Essentially the idea is, this test should also act as a good README to understanding the error messages.
Also now you can have one more test that gives out the pretty print version (may be, or may not be relevant)

Reply:
> Let's test whether index associated with errors separately.

how can I do that? 
I found only 2 possibilities to assert that case:
```
assert(res)(isLeft(equalTo(expected)))
```
or 
```
assert(res)(isLeft(hasField("size", _.size, equalTo(expected.size))))
```